lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
Actually, @zeroedin-bill that PGSQL_LIBPQ_VERSION suggestion I don't think is valid either. This is a PHP variable. My PHP (web servers) are not on the same as PGSQL (db servers).

The PGSQL_LIBPQ_VERSION constant on my web servers is going to reflect the installed pgsql client -- NOT the version of my actual server.

For example, if I have a RHEL 5 web server -- it's going to have postgresq 8.1 by default. BUT, I have PG 9.3 on my db server. Using that constant is going to make the script think I am connecting to 8.1 which is totally false.

You can't assume the client on the web server matches the database server version... right?

I'm not confused am I?